Fixed bug where some launced apps did not show up under recent/most

seiflotfy opened the pull request:
This is due to a wrong matching, matching the executable with
the .desktop file. Now it is comparing only .desktop files.

kalfa commented on panels/applications/src/dawati-netbook-launcher.c:
```diff
@@ -578,19 +580,23 @@ static void mnb_launcher_monitor_cb        (MnbLauncherMonitor *monitor,
 {
+  GSList *iter;
+  char *exec;
+
```
useless blank line here, below there are still declarations

kalfa commented on panels/applications/src/dawati-netbook-launcher.c:
```diff
@@ -578,19 +580,23 @@ static void mnb_launcher_monitor_cb        (MnbLauncherMonitor *monitor,
           clutter_actor_hide (CLUTTER_ACTOR (button));
     }
+
+  g_free(exec);
```
if the loops iterates N (>1) times, you are leaking N-1 string allocations.

seiflotfy committed:
applications: fix code styling

seiflotfy committed:
applications: fix logical bug

Fixed bug where some launched apps did not show up
under recent/most. This is due to a wrong matching, matching the
executable with the .desktop file. Now it is comparing only
.desktop files.

seiflotfy committed:
applicaitons: fixed leak

seiflotfy committed:
applications: remove unused declaration
